### DISCUSSION TITLE
Accessing data "globally"

### DISCUSSION THREAD

**Discussion by Fabio53443** [Budibase/budibase#14430] [Help] [answered]
Hi! i'm new to Budibase and i'm trying to build an app I've already built in AppSheet to see how it fares. 
I'm running into trouble undestanding how to get to data from around the app. Let me explain: 
I have two tables: a `customers` table and a `orders` table. I'm building a form for the orders table and i want to set the initial value for the "Customer" field to the last row of the customers table, but i can't seem to figure out how to access that data from the expression builder: it only seems to be able to see the data inside the form itself. Am i doing it wrong? It would be great to get some help. 
I'm using the built-in DB.

**Comment by mikesealey** [comment, marked as answer]
Hey @Fabio53443 

I'll take you through how I would get the last row of the customers table as a default value for the Customer field.

Firstly, I've mocked up some sample tables, `customers` and `orders`, with some columns including a relationship between the two.

Next, in the design section, I added a `Form Block` (although this walkthrough would also apply to a `Form component`, as well as a `Multi-Step Form Block`). Set the Data Source of the form to be Orders. 
![image](https://github.com/user-attachments/assets/22b749c5-7980-43cf-ad0f-b31cce1389f1)

I also added a Data Provider on the screen, and set the Data Source to be the customers table. Set the limit to 1, and disable pagination. Add a Headline component to the Data Provider, and set the value to be the binding of `{{ New Data Provider.Rows.0.name }}`. This will help you adjust the settings to of the Data Provider to make sure you're getting the last row of the Customers table - you may need to tweak the Ascending/Descending setting, as well as which column you want to sort by.
![image](https://github.com/user-attachments/assets/76e44bb9-2092-483c-8469-1aa83673d56c)
![image](https://github.com/user-attachments/assets/86419dda-8449-4b90-8970-31e9cad30ba4)

NB: The reason we chose to reference the customer name here is so that it's easier to debug. For your specific use case you may want to sort on an auto-column of `created_at`.

Once you're happy with the Data Provider's settings, click the little cog-icon next to your customer field. You can set the default value of the customer field in the form to be what you picked out in the Data Provider, except instead of referencing the customer's name, you'll want the _id - `{{ New Data Provider.Rows.0._id }}`
![image](https://github.com/user-attachments/assets/5794071b-183a-4a67-9b02-fcbf984baa72)

Once you're happy with the Data Provider fetching the last customer-row in the table you can delete the Headline component from the Data Provider, but don't delete the data provider itself. 

Here's the app-export where I worked through this solution for you, download it and have a closer look if it helps. You can see more about how to import an app [here](https://docs.budibase.com/docs/embedded-app).
[14430-export-1724343457919.tar.gz](https://github.com/user-attachments/files/16714853/14430-export-1724343457919.tar.gz)

**Reply by Fabio53443** [reply]
Hi! thanks for the writeup. As i was looking around i ended up adding a data provider and i wrote this js function which seems to work: 

```js
a = $("Venditori.Rows")
return a.at(-1)
```

**Reply by Fabio53443** [reply]
So sorry to bother you again, @mikesealey, but i had a quick question: I'm running a secondhand book market so i have these tables: 
1. "books" which includes ISBNs, titles, authors and prices of all the books we could sell; 
2. "inventory" which helps us keep track of all the stock we have, with a customer number and a ISBN column (which is in a many-to-one relationship with the books table)
3. customers (with all the customer data and so on).

In the inventory form, I would like to be able to enter the ISBN using the scanner feature, but as far as i can tell i can't do so unless i select "QR Code" as the column type which i would not like to do to be able to use the relationship feature with the other table. 

Is there any way i can mark a column as scannable no matter what type it is? Ideally it would search among the other table and let me select it if it exist. Thank you! let me know if you would rather i move this to its own topic.
